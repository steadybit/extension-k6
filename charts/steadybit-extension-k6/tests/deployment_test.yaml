templates:
  - deployment.yaml
tests:
  - it: manifest should match snapshot without TLS
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with TLS
    set:
      tls:
        server:
          certificate:
            fromSecret: server-cert
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with mutual TLS
    set:
      tls:
        server:
          certificate:
            fromSecret: server-cert
        client:
          certificates:
            fromSecrets:
              - client-cert-a
              - client-cert-a
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with existing secret
    set:
      k6:
        cloudApiKey: null
        existingSecret: existing-secret
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with api key
    set:
      k6:
        cloudApiKey: 111-222-333
        existingSecret: null
    asserts:
      - matchSnapshot: { }
