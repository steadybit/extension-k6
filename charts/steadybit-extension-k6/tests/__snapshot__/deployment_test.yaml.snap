manifest should match snapshot with TLS:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-k6
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-k6
      template:
        metadata:
          labels:
            app.kubernetes.io/name: steadybit-extension-k6
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/extension/certificates/server-cert/tls.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/extension/certificates/server-cert/tls.key
              image: ghcr.io/steadybit/extension-k6:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8088
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8088
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
                - mountPath: /etc/extension/certificates/server-cert
                  name: certificate-server-cert
                  readOnly: true
          volumes:
            - emptyDir: {}
              name: tmp-dir
            - name: certificate-server-cert
              secret:
                optional: false
                secretName: server-cert
manifest should match snapshot with api key:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-k6
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-k6
      template:
        metadata:
          labels:
            app.kubernetes.io/name: steadybit-extension-k6
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_CLOUD_API_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: cloud-api-token
                      name: steadybit-extension-k6
              image: ghcr.io/steadybit/extension-k6:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8088
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8088
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
          volumes:
            - emptyDir: {}
              name: tmp-dir
manifest should match snapshot with existing secret:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-k6
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-k6
      template:
        metadata:
          labels:
            app.kubernetes.io/name: steadybit-extension-k6
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_CLOUD_API_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: cloud-api-token
                      name: existing-secret
              image: ghcr.io/steadybit/extension-k6:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8088
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8088
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
          volumes:
            - emptyDir: {}
              name: tmp-dir
manifest should match snapshot with extra env vars:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-k6
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-k6
      template:
        metadata:
          labels:
            app.kubernetes.io/name: steadybit-extension-k6
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: FOO
                  value: bar
              envFrom:
                - configMapRef: null
                  name: env-configmap
                - name: env-secrets
                  secretRef: null
              image: ghcr.io/steadybit/extension-k6:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8088
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8088
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
          volumes:
            - emptyDir: {}
              name: tmp-dir
manifest should match snapshot with extra volumes:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-k6
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-k6
      template:
        metadata:
          labels:
            app.kubernetes.io/name: steadybit-extension-k6
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
              image: ghcr.io/steadybit/extension-k6:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8088
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8088
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
                - mountPath: /foobar
                  name: example
          volumes:
            - emptyDir: {}
              name: tmp-dir
            - emptyDir: {}
              name: example
manifest should match snapshot with mutual TLS:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-k6
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-k6
      template:
        metadata:
          labels:
            app.kubernetes.io/name: steadybit-extension-k6
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/extension/certificates/server-cert/tls.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/extension/certificates/server-cert/tls.key
                - name: STEADYBIT_EXTENSION_TLS_CLIENT_CAS
                  value: /etc/extension/certificates/client-cert-a/tls.crt,/etc/extension/certificates/client-cert-a/tls.crt
              image: ghcr.io/steadybit/extension-k6:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8088
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8088
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
                - mountPath: /etc/extension/certificates/client-cert-a
                  name: certificate-client-cert-a
                  readOnly: true
                - mountPath: /etc/extension/certificates/server-cert
                  name: certificate-server-cert
                  readOnly: true
          volumes:
            - emptyDir: {}
              name: tmp-dir
            - name: certificate-client-cert-a
              secret:
                optional: false
                secretName: client-cert-a
            - name: certificate-server-cert
              secret:
                optional: false
                secretName: server-cert
manifest should match snapshot with mutual TLS using containerPaths:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-k6
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-k6
      template:
        metadata:
          labels:
            app.kubernetes.io/name: steadybit-extension-k6
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/tls/server.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/tls/server.key
                - name: STEADYBIT_EXTENSION_TLS_CLIENT_CAS
                  value: /etc/tls/ca.crt,/etc/tls/ca2.crt
              image: ghcr.io/steadybit/extension-k6:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8088
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8088
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
          volumes:
            - emptyDir: {}
              name: tmp-dir
manifest should match snapshot without TLS:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels: null
      name: RELEASE-NAME-steadybit-extension-k6
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: steadybit-extension-k6
      template:
        metadata:
          labels:
            app.kubernetes.io/name: steadybit-extension-k6
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
              image: ghcr.io/steadybit/extension-k6:latest
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /health/liveness
                  port: 8088
              name: extension
              readinessProbe:
                httpGet:
                  path: /health/readiness
                  port: 8088
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 64Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 10000
                runAsNonRoot: true
                runAsUser: 10000
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-dir
          volumes:
            - emptyDir: {}
              name: tmp-dir
